presto-main/src/main/java/com/facebook/presto/server/CoordinatorModule.java

```diff
@@ -139,6 +143,14 @@ public void configure(Binder binder)
         binder.bind(NodeTaskMap.class).in(Scopes.SINGLETON);
         newExporter(binder).export(NodeScheduler.class).withGeneratedName();
 
+        // for polling worker states for graceful shutdown
+        httpClientBinder(binder).bindHttpClient("discoveryNodeManager", ForGracefulShutdown.class)
```

Comment:
Name this "node-manager" instead of "discoveryNodeManager" since this name will show up in the configs